### PR TITLE
added move left/right images #120

### LINKED ISSUE #28: Drag to move Image in assginment preview
**Is your feature request related to a problem? Please describe.**
With #26 we can now remove images before downloading PDF. But if we generated the first image then second image and now we want first image after the second image in PDF, there is no way to do that.

**Describe the solution you'd like**
In PDF Preview Screen, If we could move the image to left and right and arrange before downloading PDF, it can solve the problem.
screenshot of PDF Preview screen from tool

### LINKED ISSUE #34: Resize the image and move in around the page.
**Is your feature request related to a problem? Please describe.**
I am having issue putting the image on the page as I can't resize the image or move it around and place it as per my choice.

### LINKED ISSUE #120: Supporting Re Ordering Images After They're generated
Can we have a feature that allows drag and drop to reorder the images 

**USE CASE**
Sometimes it may happen that we need this because you may make a mistake in one page and create many pages after it
There is no way to undo other than deleting all of the other images

### PR DESCRIPTION
**Pull request by StTronn**
Hey main changes are in `generate-images.mjs` added `moveLeft` and `moveRight` function. Let me know If some changes are required.

Edit:
Fixes #120 
Fixes #28 
Fixes #34

### CODE REVIEW

**Comment by StTronn** [comment]
Hey, I am new to cypress so sorry for failing the test. Seems like might be an issue from ci. I have run the test locally they are passing 100%.

**Comment by saurabhdaware** [comment]
Yeah some issue with cypress

File: js/utils/generate-utils.mjs [hunk 1]
```diff
@@ -95,6 +95,21 @@ function renderOutput(outputImages) {
...
+          data-index="${index}" 
+        >
+          L 
+        </button>
+        <button 
+          class="button move-right"
+          data-index="${index}" 
+        >
+          R 
+        </button>
```
Comment (saurabhdaware) [inline js/utils/generate-utils.mjs]:
```suggestion
        <button 
          class="button move-left"
          data-index="${index}" 
        >
          Move Left
        </button>
        <button 
          class="button move-right"
          data-index="${index}" 
        >
          Move Right
        </button>
```

Maybe we can change L to `Move Left`

File: js/generate-images.mjs [hunk 2]
```diff
@@ -102,8 +102,30 @@ export const deleteAll = () => {
...
+    while (k--) {
+      arr.push(undefined);
+    }
+  }
+  arr.splice(newIndex, 0, arr.splice(oldIndex, 1)[0]);
+  return arr; // for testing
+};
+
+export const moveLeft = (index) => {
+  if (index === 0) return outputImages;
```
Comment (saurabhdaware) [inline js/generate-images.mjs:120]: image
 I am getting this error when I click Right on the rightmost item. Any idea what could be causing this?

**Review by saurabhdaware — CHANGES_REQUESTED** [review]
Hey, this looks and works great! 

Just mentioned some small changes. Let me know if you would like to do it :D

Thanks for the PR!

File: js/generate-images.mjs (hunk 2 above)
Comment (StTronn) [reply js/generate-images.mjs:120]: Hey seems I did an off-by-one error `moveRight` function. Will correct it

File: js/utils/generate-utils.mjs (hunk 1 above)
Comment (StTronn) [reply js/utils/generate-utils.mjs]: Yeah makes sense doing it right away :)

**Comment by StTronn** [comment]
Hi, Saurabh sorry I did the changes the same day but forgot to ping 😅  I have made those changes have a look and let me know of other changes :)

**Review by saurabhdaware — APPROVED** [review]
Looks good @StTronn. Thank you for contributing 🥳

**Pull request merged**